main.tf

```diff
@@ -135,7 +135,9 @@ resource "google_container_cluster" "domino_cluster" {
   # separately managed node pools. So we create the smallest possible default
   # node pool and immediately delete it.
   remove_default_node_pool = true
-  initial_node_count       = 1
+
+  # Workaround for https://github.com/terraform-providers/terraform-provider-google/issues/3385
+  initial_node_count = max(1, var.compute_nodes_min) + max(0, var.gpu_nodes_min) + var.platform_nodes_max
```

Comment:
I think this may actually exacerbate the issue since there are now more machines that need to be removed from the default node pool. Have you iteratively tested this?

Reply:
Yeah, I think the tradeoff is that it might cause a bit more time initially for a more reliably "complete" provision at the end of the terraform run. Do you mean iterative number of initial machines or whether this actually fixes the issue (it did appear to, but I'll do some more rigorous testing)?

Reply:
It may result in longer provisioning time but I think the intention is that, if we already know how many nodes our other pools will have initially, then setting this value will ensure the cluster does not report ready until the "full replacement" of the default node pool is complete.

It's unfortunate that we have to create a larger default node pool to accomplish this but... 💩 🤷‍♂

Reply:
My read of the linked issue is that it's actually resizing the _master_ cluster after the node pools are added and setting the initial node count ensures that it's the correct size-ish from the beginning.

Reply:
Yes, the https://github.com/terraform-providers/terraform-provider-google/issues/3385#issuecomment-488149996 seems to accurately describe this unexpected master upgrade behavior.